advanced_functionality/autogluon-multimodal-containers/scripts/multimodal_serve.py

```diff
@@ -0,0 +1,53 @@
+import base64
+import hashlib
+import json
+from io import BytesIO, StringIO
+
+import pandas as pd
+from PIL import Image
+
+from autogluon.multimodal import MultiModalPredictor
+
+
+def _decode_and_save_base85_image(image_b85):
+    """saves Base65 encoded image to disk and returns unique name"""
+
+    image_bytes = base64.b85decode(image_b85)
+    image_hash = hashlib.md5(image_bytes).hexdigest()
+    image_name = f'image_{image_hash}.png'
+    image = Image.open(BytesIO(image_bytes))
+    image.save(image_name)
+
+    return image_name
+
+def model_fn(model_dir):
+    """loads model from previously saved artifact"""
+
+    model = MultiModalPredictor.load(model_dir)
+    globals()["column_names"] = \
+        [column_name for column_name in model._column_types.keys() if column_name != model._label_column]
```

Comment:
@yinweisu @sxjscience do we need to send it to GPU for faster inference? If yes, what's the procedure?

Reply:
Yes, I'm interested in other's feedback. I was expecting the GPU would be used for inference here based on the config, which was trained on GPU and inference machine has GPU.  In the `config.yaml`, we have `num_gpus: -1` and `auto_select_gpus: true`. Unfortunately I can't find good logs from SageMaker on whether GPU was used, but the logs do show 1 GPU being detected.

Reply:
Multimodal should use gpu for inference by default if it detects it. No special things need to be done